test/e2e/set/scyllacluster/scyllacluster_expose.go

```diff
@@ -4,26 +4,258 @@ package scyllacluster
+					Type: scyllav1.NodeServiceTypeClusterIP,
+				},
+				BroadcastOptions: &scyllav1.NodeBroadcastOptions{
+					Nodes: scyllav1.BroadcastOptions{
+						Type: scyllav1.BroadcastAddressTypeServiceClusterIP,
+					},
+					Clients: scyllav1.BroadcastOptions{
+						Type: scyllav1.BroadcastAddressTypePodIP,
+						PodIP: &scyllav1.PodIPAddressOptions{
+							Source: scyllav1.StatusPodIPSource,
+						},
+					},
+				},
+			},
+			validateService: func(svc *corev1.Service) {
+				o.Expect(svc.Spec.Type).To(o.Equal(corev1.ServiceTypeClusterIP))
+				parsedClusterIP := net.ParseIP(svc.Spec.ClusterIP)
+				o.Expect(parsedClusterIP).ToNot(o.BeNil())
+			},
+			validateScyllaConfig: func(ctx context.Context, configClient *scyllaclient.ConfigClient, svc *corev1.Service, pod *corev1.Pod) {
+				broadcastAddress, err := configClient.BroadcastAddress(ctx)
+				o.Expect(err).NotTo(o.HaveOccurred())
+
+				o.Expect(broadcastAddress).To(o.Equal(svc.Spec.ClusterIP))
+
+				broadcastRPCAddress, err := configClient.BroadcastRPCAddress(ctx)
+				o.Expect(err).NotTo(o.HaveOccurred())
+				o.Expect(broadcastRPCAddress).To(o.Equal(pod.Status.PodIP))
+			},
+		}),
+		g.Entry("using Headless Service, and broadcast PodIP from Pod Status to both client and nodes", &entry{
+			exposeOptions: &scyllav1.ExposeOptions{
+				NodeService: &scyllav1.NodeServiceTemplate{
+					Type: scyllav1.NodeServiceTypeHeadless,
+				},
+				BroadcastOptions: &scyllav1.NodeBroadcastOptions{
+					Nodes: scyllav1.BroadcastOptions{
+						Type: scyllav1.BroadcastAddressTypePodIP,
+						PodIP: &scyllav1.PodIPAddressOptions{
+							Source: scyllav1.StatusPodIPSource,
+						},
+					},
+					Clients: scyllav1.BroadcastOptions{
+						Type: scyllav1.BroadcastAddressTypePodIP,
+						PodIP: &scyllav1.PodIPAddressOptions{
+							Source: scyllav1.StatusPodIPSource,
+						},
+					},
+				},
+			},
+			validateService: func(svc *corev1.Service) {
+				o.Expect(svc.Spec.Type).To(o.Equal(corev1.ServiceTypeClusterIP))
+				o.Expect(svc.Spec.ClusterIP).To(o.Equal(corev1.ClusterIPNone))
+			},
+			validateScyllaConfig: func(ctx context.Context, configClient *scyllaclient.ConfigClient, svc *corev1.Service, pod *corev1.Pod) {
+				broadcastAddress, err := configClient.BroadcastAddress(ctx)
+				o.Expect(err).NotTo(o.HaveOccurred())
+
+				o.Expect(broadcastAddress).To(o.Equal(pod.Status.PodIP))
+
+				broadcastRPCAddress, err := configClient.BroadcastRPCAddress(ctx)
+				o.Expect(err).NotTo(o.HaveOccurred())
+				o.Expect(broadcastRPCAddress).To(o.Equal(pod.Status.PodIP))
+			},
+		}),
+		g.Entry("using LoadBalancer Service, and broadcast external IP to client and PodIP to nodes", &entry{
+			exposeOptions: &scyllav1.ExposeOptions{
+				NodeService: &scyllav1.NodeServiceTemplate{
+					Type: scyllav1.NodeServiceTypeLoadBalancer,
+					// Change to non-default LB class to avoid conflicts on status when running in cloud
+					LoadBalancerClass: pointer.Ptr("my-lb-class"),
```

Comment:
nit: If it truly doesn't matter, can it be a random string so it runs inside customer clusters where an ingress class named `my-lb-class` is not impossible?

Reply:
done